src/ui_iteminfo.cpp

```diff
@@ -0,0 +1,153 @@
+                            }
+                            if( iVal > kVal - .01 &&
+                                iVal < kVal + .01 ) {
+                                this_color = c_light_gray;
+                            } else if( iVal > kVal ) {
+                                if( i.bLowerIsBetter ) {
+                                    this_color = c_light_red;
+                                } else {
+                                    this_color = c_light_green;
+                                }
+                            } else if( iVal < kVal ) {
+                                if( i.bLowerIsBetter ) {
+                                    this_color = c_light_green;
+                                } else {
+                                    this_color = c_light_red;
+                                }
+                            }
+                            break;
+                        }
+                    }
+                }
+                formatted_string += colorize( i.sValue, this_color );
+            }
+            if( pos != std::string::npos ) {
+                formatted_string += sFmt.substr( pos + 5 );
+            }
+            window.draw_colored_text( formatted_string, base_color, wrap_width );
+            same_line = !i.bNewLine;
+        }
+    }
+}
+
+iteminfo_window::iteminfo_window( item_info_data &info, point pos, int width, int height,
+                                  ImGuiWindowFlags flags ) :
+    cataimgui::window( "iteminfo", flags ),
+    data( info ), pos( pos ), width( width ), height( height )
+{
+    ctxt = input_context( "default" );
+    if( info.handle_scrolling ) {
+        ctxt.register_action( "PAGE_UP" );
+        ctxt.register_action( "PAGE_DOWN" );
+    }
+    ctxt.register_action( "CONFIRM" );
+    ctxt.register_action( "QUIT" );
+    if( info.any_input ) {
+        ctxt.register_action( "ANY_INPUT" );
+    }
+
+    ctxt.set_timeout( 10 );
+}
+
+cataimgui::bounds iteminfo_window::get_bounds()
+{
+    return { static_cast<float>( str_width_to_pixels( pos.x ) ),
+             static_cast<float>( str_height_to_pixels( pos.y ) ),
+             static_cast<float>( str_width_to_pixels( width ) ),
+             static_cast<float>( str_height_to_pixels( height ) ) };
+}
+
+void iteminfo_window::draw_controls()
+{
+    draw_item_info_imgui( *this, data, width );
+
+    if( scroll != 0 ) {
+        float scroll_pos = ImGui::GetScrollY() + scroll *
+                           ImGui::GetCurrentContext()->CurrentWindow->InnerRect.GetHeight();
+        ImGui::SetScrollY( scroll_pos );
+        scroll = 0;
+    }
+}
+
+void iteminfo_window::execute()
+{
+    if( data.without_getch ) {
+        ui_manager::redraw();
+        return;
+    }
+
+    while( true ) {
+        ui_manager::redraw();
+        std::string action = ctxt.handle_input();
+
+        if( data.handle_scrolling && action == "PAGE_UP" ) {
+            scroll = -1;
```

Comment:
Of course, if you change `scroll` to be absolute instead of relative then this needs bounds checking.

Reply:
And after trying it out, I really miss not being able to scroll with the arrow keys.

Reply:
Did I miss that somehow, was arrow key scrolling a thing in the original UI?

Reply:
No, but I definitely miss it. It’s such a natural thing to want to do, but using the arrow keys just closes the window.